zombies.js

```diff
@@ -78,137 +94,210 @@
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+  this.discardItem = function (item) {
+    let indexOfItem = this.getPack().indexOf(item);
+    if (this.getPack().includes(item)) {
+      this.getPack().splice(indexOfItem, 1);
+      console.log(this.name + ' discarded ' + this.item);
+      return true;
+    } else {
+      console.log('item not found');
+      return false;
+    }
+  }
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  this.equip = function (itemToEquip) {
+    if (itemToEquip instanceof Weapon && this.getPack().includes(itemToEquip)) {
+      let indexOfItemTo = this.getPack().indexOf(itemToEquip);
+      if (this.equipped) {
+        this.getPack().splice(indexOfItemTo, 1, this.equipped);
+        this.equipped = itemToEquip;
+      } else {
+        this.getPack().splice(indexOfItemTo, 1)
+        this.equipped = itemToEquip;
+      }
+    }
+  }
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+  this.eat = function (itemToEat) {
+    if (itemToEat instanceof Food && this.getPack().includes(itemToEat)) {
+      let indexOfItemToEat = this.getPack().indexOf(itemToEat);
+      this.getPack().splice(indexOfItemToEat, 1);
+      if (itemToEat.energy + this.health < this.getMaxHealth()) {
+        this.health += itemToEat.energy;
```

Comment:
You are doing the same work twice. A better approach might be to apply the new health and then test if it above max, if so, lower to max.

```js
this.health += itemToEat.energy;
if(this.health > this.getMaxHealth()) {
  this.health = this.getMaxHealth();
}
```